src/simulation/estimation_setup/processTrackingTxtFile.cpp

```diff
@@ -0,0 +1,169 @@
+      break;
+    }
+    default: {
+      throw std::runtime_error("Error while processing tracking txt file: Time representation not recognised or implemented.");
+    }
+  }
+}
+
+void ProcessedTrackingTxtFileContents::updateLinkEnds()
+{
+  linkEndsVector_.clear();
+  const auto& dataMap = rawTrackingTxtFileContents_->getDoubleDataMap();
+  const auto& metaDataStrMap = rawTrackingTxtFileContents_->getMetaDataStrMap();
+  const auto& numDataRows = rawTrackingTxtFileContents_->getNumRows();
+  LinkEndsRepresentation linkEndsRepresentation = getLinkEndsRepresentation();
+
+  switch (linkEndsRepresentation) {
+
+    // TODO: make a cleaner implementation to allow adding different ways of providing the link ends easily
+    case dsn_transmitting_receiving_station_nr: {
+      const auto& dsnTransmitterIds = dataMap.at(input_output::TrackingDataType::dsn_transmitting_station_nr);
+      const auto& dsnReceiverIds = dataMap.at(input_output::TrackingDataType::dsn_receiving_station_nr);
+
+      for (size_t i = 0; i < numDataRows; ++i) {
+        LinkEnds currentLinkEnds{
+            {transmitter, LinkEndId("Earth", getStationNameFromStationId("DSS-", static_cast<int>(dsnTransmitterIds[i])))},
+            {reflector, LinkEndId(spacecraftName_, "Antenna")},
+            {receiver, LinkEndId("Earth", getStationNameFromStationId("DSS-", static_cast<int>(dsnReceiverIds[i])))},
+        };
+        linkEndsVector_.push_back(currentLinkEnds);
+      }
+      break;
+    }
+
+    case vlbi_station: {
+
+      // TODO: Different if in the metadata!
+      //    Take data from actual info, not hardcoded!
+
+      if (metaDataStrMap.count(input_output::TrackingDataType::vlbi_station_name)) {
+        std::string vlbi_station_name = metaDataStrMap.at(input_output::TrackingDataType::vlbi_station_name);
+        LinkEnds constantLinkEnds{
+            {transmitter, LinkEndId(spacecraftName_, "Antenna")},
+            {receiver, LinkEndId("Earth", vlbi_station_name)}, // FIXME!
+        };
+        for (size_t i = 0; i < numDataRows; ++i) {
+          linkEndsVector_.push_back(constantLinkEnds);
+        }
+      } else if (dataMap.count(input_output::TrackingDataType::dsn_receiving_station_nr)) {
+        for (size_t i = 0; i < numDataRows; ++i) {
+          std::string vlbi_station_name = metaDataStrMap.at(input_output::TrackingDataType::vlbi_station_name);
+          LinkEnds currentLinkEnds{
+              {transmitter, LinkEndId(spacecraftName_, "Antenna")},
+              {receiver, LinkEndId("Earth", vlbi_station_name)}, // FIXME!
+          };
+          linkEndsVector_.push_back(currentLinkEnds);
+        }
+      }
+      break;
+    }
+
+    default: {
+      throw std::runtime_error("Error while processing tracking txt file: LinkEnds representation not recognised or implemented.");
+    }
+  }
+
+  // Creating a set with all the various linkEnds
+  linkEndsSet_ = vectorToSet(linkEndsVector_);
```

Comment:
why does this have to be a set rather than a vector?

Reply:
This remained from a first implementation and is no longer used anywhere. I will remove it.

Reply:
```
11 15 15 2011 APR 04 19:29:00    629786947.906   0
11 55 55 2011 APR 05 17:02:18    619032920.847   0
11 25 25 2011 APR 05 18:53:48    618145818.935  77
11 55 55 2011 APR 06 16:40:49    608667943.676   0
11 26 26 2011 APR 08 19:03:22    591649291.183  77
```

The above are some consecutive readings in https://ssd.jpl.nasa.gov/dat/planets/messenger.txt. The second and third column are DSN stations. So the observations have to be reordered for the collection. If there is already functionality in the `Observationcollection`, I can certainly adjust it.

Reply:
Right, now I remember :) Perfectly good solution. Leave as it is.